device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsExceptionTranslator.java

```diff
@@ -14,6 +14,13 @@
 {
     static TransportException convertFromAmqpException(ErrorCondition error)
     {
+        if (error == null)
+        {
+            TransportException t = new TransportException("An unknown transport exception occurred");
+            t.setRetryable(true);
```

Comment:
shouldn't we mark an unknown error code to not be retryable?

Reply:
That isn't the current behavior. Down below, if the error code doesn't match any known code, we return the same exception